neqo-http3/src/connection_client.rs

```diff
@@ -5930,4 +5945,27 @@ mod tests {
         };
         assert!(server.conn.events().any(stop_sending_event));
     }
+
+    fn do_test_client_critical_stream_create_failed(max_stream: u64) {
```

Comment:
I think that this is a case where a generic function might be better.  That is, it might be better to have these two functions shaped like this:

```rust
#[test]
fn client_control_stream_create_fail() {
       let mut client = default_http3_client();
       let mut server = TestServer::new();
       server.set_max_uni_stream(max_stream);
       handshake_client_error(&mut client, &Error::StreamLimitError);
}
```

What do you think?  It's a little more duplicated code (you run the setup steps twice), but it means that we get a generic "handshake breaks" handler.

Reply:
Yeah, I agree this looks clearer.
My origin code `do_test_client_critical_stream_create_failed(0);` does not look easy to understand.